### PR TITLE
BOAC-1983: Marks updated or newly created note as read by user who did the deed

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1983

### CODE REVIEW

File: boac/merged/advising_note.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def note_to_compatible_json(note, topics=None, attachments=None):
         'createdAt': _resolve_created_at(note),
         'updatedBy': note.get('updated_by'),
         'updatedAt': _resolve_updated_at(note),
-        'read': False,
+        'read': True if note_read else False,
```
Comment (johncrossman) [inline boac/merged/advising_note.py]: @lyttam Can `True if note_read else False` be replaced with `note_read`?

File: boac/api/notes_controller.py [hunk 2]
```diff
@@ -90,7 +91,10 @@ def update_note():
         subject=subject,
         body=body,
     )
-    note_json = note_to_compatible_json(note.to_api_json())
+    note_read = NoteRead.is_note_read_by_user(current_user.id, note.id)
+    if note and not note_read:
+        note_read = NoteRead.create(current_user.id, note.id)
+    note_json = note_to_compatible_json(note.to_api_json(), note_read=note_read)
```
Comment (johncrossman) [inline boac/api/notes_controller.py]: @lyttam Minor concern: Perhaps `is_note_read_by_user()` is not needed? I.e., no harm in always invoking `NoteRead.create(...)`.  (Also, no null check on `note` is necessary.)

File: boac/merged/advising_note.py (hunk 1 above)
Comment (lyttam) [reply boac/merged/advising_note.py]: @johncrossman Tried that, but `note_read` is an object that's not JSON serializable. I could probably implement a method on `NoteRead` to make it serializable, if you think it's worth it.

File: boac/api/notes_controller.py (hunk 2 above)
Comment (lyttam) [reply boac/api/notes_controller.py]: @johncrossman you're right, I'll remove the null check.  But I found that calling `NoteRead.create` twice with the same arguments resulted in a database error (primary key already exists). Should I dispense with `NoteRead.is_note_read_by_user` and add some protection to `NoteRead.create` instead?

File: boac/merged/advising_note.py (hunk 1 above)
Comment (johncrossman) [reply boac/merged/advising_note.py]: Not necessary, @lyttam This works.

**Comment by lyttam** [comment]
@johncrossman alright, ready for re-review.

**Pull request merged**